### PR TITLE
Joe's HW7 file

### PR DESCRIPTION
**Pull request by puleo-joe**
@MengyuanZoe @hanghu @YangJiang89 My HW7 file for editing.

### CODE REVIEW

File: HW7/JPHW7.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#import data manipulators
+import pandas as pd
+
+#import folium plugins
+from folium.plugins import MarkerCluster
+import folium
+from folium import plugins
+
+#load King County datasets
+crimes = pd.read_csv('King_County_Sheriff_s_Office.csv')
```
Comment (hanghu) [inline HW7/JPHW7.py]: the data is not in the same folder, you might want to add the path of the data files.

File: HW7/JPHW7.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+'''Populating the map with clusters of houses and a heat map of crime rates.'''
+
+#import system modules
+import os
+
+#import data manipulators
+import pandas as pd
+
+#import folium plugins
+from folium.plugins import MarkerCluster
```
Comment (hanghu) [inline HW7/JPHW7.py:9]: unused modules MarkerCluster and os

File: HW7/JPHW7.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		price: float64
+		size: int64
+		bedrooms: 1nt64
+		bathrooms: float64
+		year built: int64'''
+
+for name, row in houses[0:200].iterrows():
+    folium.Marker([row["lat"], row["long"]],
+                  popup="Price: {0}$, Size: {1} sqft, Bedrooms: {2}, Bathrooms: {3}, Year Built: {4}"\
+                  .format(row["price"], row['sqft_living'], row['bedrooms'], row['bathrooms'], row['yr_built'])).add_to(house_groups)
```
Comment (hanghu) [inline HW7/JPHW7.py]: the above two lines are too long

File: HW7/JPHW7.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+'''Create new basemap for crimes heat map'''
+kccrimes_heatmap = folium.Map(location = [houses['lat'].mean(), houses['long'].mean()], zoom_start = 10)
+
+'''Add data to heat map and select opacity'''
+kccrimes_heatmap.add_children(
+    plugins.HeatMap([[row["latitude"], row["longitude"],row["city"]]
+    for iters, row in crimes[0:100].iterrows()],
+    min_opacity=1, max_zoom=10, radius = 8))
+
```
Comment (hanghu) [inline HW7/JPHW7.py:48]: The indentation of above two lines are wrong

File: HW7/JPHW7.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+for name, row in houses[0:200].iterrows():
+    folium.Marker([row["lat"], row["long"]],
+                  popup="Price: {0}$, Size: {1} sqft, Bedrooms: {2}, Bathrooms: {3}, Year Built: {4}"\
+                  .format(row["price"], row['sqft_living'], row['bedrooms'], row['bathrooms'], row['yr_built'])).add_to(house_groups)
+
+#save output of house markers
+kingcounty_map.save('houses_map.html')
+
+'''Create new basemap for crimes heat map'''
```
Comment (hanghu) [inline HW7/JPHW7.py:40]: for one line comments, a simple "#" before the comments could work too

File: HW7/JPHW7.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+kccrimes_heatmap = folium.Map(location = [houses['lat'].mean(), houses['long'].mean()], zoom_start = 10)
```
Comment (YangJiang89) [inline HW7/JPHW7.py]: no white space around '=' as the function parameter argument

File: HW7/JPHW7.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+from folium import plugins
+
+#load King County datasets
+crimes = pd.read_csv('King_County_Sheriff_s_Office.csv')
+houses = pd.read_csv("kc_house_data.csv", parse_dates=['date'])
+
+#prepare base map
+'''Use mean of latitude and longitude in dataset and set original zoom'''
+kingcounty_map = folium.Map(location = [houses['lat'].mean(),
+	houses['long'].mean()], zoom_start = 10)
```
Comment (YangJiang89) [inline HW7/JPHW7.py]: Wrong continued indentation

File: HW7/JPHW7.py (hunk 6 above)
Comment (YangJiang89) [inline HW7/JPHW7.py]: Line too long

File: HW7/JPHW7.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+#save output of house markers
+kingcounty_map.save('houses_map.html')
+
+'''Create new basemap for crimes heat map'''
+kccrimes_heatmap = folium.Map(location = [houses['lat'].mean(), houses['long'].mean()], zoom_start = 10)
+
+'''Add data to heat map and select opacity'''
+kccrimes_heatmap.add_children(
+    plugins.HeatMap([[row["latitude"], row["longitude"],row["city"]]
```
Comment (YangJiang89) [inline HW7/JPHW7.py:45]: white space after comma

File: HW7/JPHW7.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    for iters, row in crimes[0:100].iterrows()],
+    min_opacity=1, max_zoom=10, radius = 8))
```
Comment (YangJiang89) [inline HW7/JPHW7.py]: no white space around '=' as the function parameter argument

**Pull request merged**